app/controllers/user_notifications_controller.rb

```diff
@@ -0,0 +1,36 @@
+class UserNotificationsController < Api::V2::BaseController
+  before_action :require_login
+  before_action :find_resource, :only => [:update, :destroy]
+
+  def index
+    @notifications = NotificationRecipient.
+    where(:user_id => User.current.id).
+    order(:created_at).
+    eager_load(:notification, :notification_type)
+
+    render :json => {
+      :notifications => @notifications.paginate(paginate_options).map(&:payload),
+      :total => @notifications.count
+    }
+  end
+
+  def update
+    # only allowed attribute to change is 'seen'
+    process_response @user_notification.update_attribute(:seen, params[:user_notification][:seen])
```

Comment:
I don't think this works already, as the params have not been permitted here. I'll fix that on my update. Thanks for the catch 😄